cpp/src/utilities/column_utils.cuh

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */ 
+
+#ifndef __UTILITIES_COLUMN_UTILS_CUH__
+#define __UTILITIES_COLUMN_UTILS_CUH__
+
+#include <cudf/types.h>
+#include <utilities/error_utils.hpp>
+#include <utilities/bit_util.cuh>
+#include <utilities/cuda_utils.hpp>
+#include <bitmask/legacy/bit_mask.cuh>
+
+using bit_mask::bit_mask_t;
+
+namespace cudf {
+
+constexpr int warp_size = 32;
+
+template <bool source_mask_valid, typename BitContainer, typename Predicate, typename Size>
+__global__ void null_if_kernel(
+    const BitContainer* source_mask, 
+    BitContainer* destination_mask, 
+    Predicate p,
+    Size num_bits
+  ){
+ 
+  static_assert(warp_size == util::size_in_bits<BitContainer>(), 
+      "warp size is different from BitConstainer size.");
+
+  Size bit_index = threadIdx.x + blockIdx.x * blockDim.x;
+
+  BitContainer active_threads =
+      __ballot_sync(0xffffffff, bit_index < num_bits);
+
+  while (bit_index < num_bits) {
+    
+    bool const predicate_is_true = p(bit_index);
+    // This function name is `null_if` so if predicate is true the bit is to be set to off
```

Comment:
See my comment about `valid_if`, so you don't have to negate the predicate. Negating the predicate is confusing...

Reply:
Yeah, that's true. I originally asked for `null_if`, but I got my logic mixed up and agree that `valid_if` is a better name.